cloudinit/config/cc_ca_certs.py

```diff
@@ -44,60 +44,110 @@
+                                                cfg['ca_cert_filename'])
+    else:
+        cfg = {}
+    return cfg
 
 
-def update_ca_certs():
+def update_ca_certs(distro_name):
     """
     Updates the CA certificate cache on the current machine.
     """
-    subp.subp(["update-ca-certificates"], capture=False)
+    distro_cfg = _distro_ca_certs_configs(distro_name)
+    subp.subp([distro_cfg['ca_cert_update_exe']], capture=False)
 
 
-def add_ca_certs(certs):
+def add_ca_certs(distro_name, certs):
     """
     Adds certificates to the system. To actually apply the new certificates
     you must also call L{update_ca_certs}.
 
     @param certs: A list of certificate strings.
     """
+    distro_cfg = _distro_ca_certs_configs(distro_name)
     if certs:
         # First ensure they are strings...
         cert_file_contents = "\n".join([str(c) for c in certs])
-        util.write_file(CA_CERT_FULL_PATH, cert_file_contents, mode=0o644)
-
-        if os.stat(CA_CERT_CONFIG).st_size == 0:
-            # If the CA_CERT_CONFIG file is empty (i.e. all existing
-            # CA certs have been deleted) then simply output a single
-            # line with the cloud-init cert filename.
-            out = "%s\n" % CA_CERT_FILENAME
-        else:
-            # Append cert filename to CA_CERT_CONFIG file.
-            # We have to strip the content because blank lines in the file
-            # causes subsequent entries to be ignored. (LP: #1077020)
-            orig = util.load_file(CA_CERT_CONFIG)
-            cur_cont = '\n'.join([line for line in orig.splitlines()
-                                  if line != CA_CERT_FILENAME])
-            out = "%s\n%s\n" % (cur_cont.rstrip(), CA_CERT_FILENAME)
-        util.write_file(CA_CERT_CONFIG, out, omode="wb")
+        util.write_file(distro_cfg['ca_cert_full_path'],
+                        cert_file_contents,
+                        mode=0o644)
+
+        if distro_cfg['ca_cert_config'] != '':
+            if os.stat(distro_cfg['ca_cert_config']).st_size == 0:
+                # If the CA_CERT_CONFIG file is empty (i.e. all existing
+                # CA certs have been deleted) then simply output a single
+                # line with the cloud-init cert filename.
+                out = "%s\n" % distro_cfg['ca_cert_filename']
+            else:
+                # Append cert filename to CA_CERT_CONFIG file.
+                # We have to strip the content because blank lines in the file
+                # causes subsequent entries to be ignored. (LP: #1077020)
+                orig = util.load_file(distro_cfg['ca_cert_config'])
+                cr_cont = '\n'.join([line for line in orig.splitlines()
+                                    if line != distro_cfg['ca_cert_filename']])
+                out = "%s\n%s\n" % (cr_cont.rstrip(),
+                                    distro_cfg['ca_cert_filename'])
+            util.write_file(distro_cfg['ca_cert_config'], out, omode="wb")
```

Comment:
This feels like it could quite naturally be refactored to a separate function; what do you think?

Reply:
current function is included add_ca_certs and update_cert_config, so I agree to separate this